File: app/src/main/java/social/entourage/android/api/model/Stats.java
package social.entourage.android.api.model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Stats implements Serializable {

    private static final long serialVersionUID = -6285877183504670635L;

    // ----------------------------------
    // ATTRIBUTES
    // ----------------------------------

    @SerializedName("tour_count")
    private int tourCount;

    @SerializedName("encounter_count")
    private int encounterCount;

    @SerializedName("entourage_count")
    private int entourageCount;

    // ----------------------------------
    // CONSTRUCTOR
    // ----------------------------------

    public Stats(int tourCount, int encounterCount, int entourageCount) {
        this.tourCount = tourCount;
        this.encounterCount = encounterCount;
        this.entourageCount = entourageCount;
    }

    // ----------------------------------
    // GETTERS & SETTERS
    // ----------------------------------

    public int getTourCount() {
        return tourCount;
    }

    public int getEncounterCount() {
        return encounterCount;
    }

    public void setTourCount(int tourCount) {
        this.tourCount = tourCount;
    }

    public void setEncounterCount(int encounterCount) {
        this.encounterCount = encounterCount;
    }

    public int getEntourageCount() {
        return entourageCount + tourCount;
    }

    public void setEntourageCount(final int entourageCount) {
        this.entourageCount = entourageCount;
    }

}

File: app/src/main/java/social/entourage/android/api/model/Stats.kt
package social.entourage.android.api.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class Stats(
    @field:SerializedName("entourage_count") private var entourageCount: Int,
    @field:SerializedName("actions_count") var actionsCount: Int,
    @field:SerializedName("events_count") var eventsCount: Int,
    @field:SerializedName("good_waves_participation") var isGoodWavesValidated: Boolean,
    @field:SerializedName("contribution_creation_count") var contribCreationCount: Int,
    @field:SerializedName("outings_count") var outingsCount: Int,
    @field:SerializedName("neighborhoods_count") var neighborhoodsCount: Int,
    @field:SerializedName("ask_for_help_creation_count") var askCreationCount: Int
) : Serializable {

    fun getActionCount(): Int {
        return entourageCount
    }

    val isEngaged: Boolean
        get() = (entourageCount > 0
                || actionsCount > 0
                || eventsCount > 0
                || isGoodWavesValidated)

    companion object {
        private const val serialVersionUID = -90028118L
    }
}